### ISSUE TITLE
Can't access files outside of jupyter root despite .lsp_symlink hack

### ISSUE THREAD

**Issue by slavistan** [jupyter-lsp/jupyterlab-lsp#828]
Hi! I use jupyter locally and work with conda environments located outside of the jupyter content dir. I ran into the problem of being unable to access files:

![image](https://user-images.githubusercontent.com/35600954/185789101-1987b6ed-4da7-4929-9112-ea3636160eab.png)

As suggested in #750 the workaround is, inside of the jupyter content dir, to create a symlink named `.lsp_symlink` to the system's root dir: `ln -s / /home/user/my_jupyter_dir/.lsp_symlink`. However, even with the symlink in place, I get the same error despite being able to access the file via the symlink:

```
$ ls -la .lsp_symlink/home/stan/.conda/envs/pytorch_cuda3090/lib/python3.10/site-packages/transformers/models/auto/tokenization_auto.py
-rw-rw-r-- 2 stan stan 33339 Jun 18 09:36 .lsp_symlink/home/stan/.conda/envs/pytorch_cuda3090/lib/python3.10/site-packages/transformers/models/auto/tokenization_auto.py
``` 

Is there anything I've missed? The instructions for the symlink hack differ between the comments in the above issue and the [installation instructions](https://github.com/jupyter-lsp/jupyterlab-lsp#installation), so I may have misunderstood something.

## Context

- Operating System and version: Arch Linux 5.19.1
- Browser and version: Brave 1.42.97.-1
- JupyterLab version: 3.4.5
- Language server and version: python-lsp-server 1.5.0

```
$ jupyter serverextension list
# config dir: /home/stan/.conda/envs/pytorch_cuda3090/etc/jupyter
#     jupyter_lsp  enabled
#     - Validating...
#       jupyter_lsp 1.5.1 OK
#     jupyterlab  enabled
#     - Validating...
#       jupyterlab 3.4.5 OK
# config dir: /etc/jupyter
#     jupyterlab  enabled
#     - Validating...
#       jupyterlab 3.4.5 OK

$ jupyter labextension list
# JupyterLab v3.4.5
# /home/stan/.conda/envs/pytorch_cuda3090/share/jupyter/labextensions
#         @jupyter-widgets/jupyterlab-manager v3.0.0 enabled OK (python, jupyterlab_widgets)
#         @krassowski/jupyterlab-lsp v3.10.1 enabled OK (python, jupyterlab-lsp)
# 
# /usr/share/jupyter/labextensions
#         jupyterlab_pygments v0.2.2 enabled OK (python, jupyterlab_pygments)
```

**Comment by slavistan** [comment]
Turns out that for the *.lsp_symlink* hack to work, hidden files must be enabled at the server level, presumably because the symlink is a hidden file ("hidden" == starting with a period character). So the instructions for *nix machines is

```
# from the jupyter project root dir
ln -s / .lsp_symlink
jupyter lab --ContentsManager.allow_hidden=True .  # or enable hidden files in the server config
```

@krassowski Would it be possible to update the documentation to add this little but essential detail to the .lsp_symlink workaround doc? And are the current instructions of symlinking /home still relevant? If you take contribution on the doc I'll gladly clarify this point.

**Comment by krassowski** [comment]
Thanks for digging into this. The instructions were fine for jupyter-server <1.17.1 but there was a breaking change in 1.17.1 fixing a security vulnerability (https://github.com/advisories/GHSA-q874-g24w-4q9g) which effectively prevents using hidden directories without explicitly enabling them, so indeed they need updating.

Would you like to open a PR?

**Comment by slavistan** [comment]
Thanks for your reply - I'll take care of it. May take a day or two, though.

**Comment by krassowski** [comment]
Closing as addressed by #846. Thanks again!
